methods/ga4gh_0.5.1/hhbeacon/app/server/beacon/BeaconV0_5_1.scala

```diff
@@ -48,10 +48,38 @@ object BeaconV0_5_1 extends BEACON {
   // PUT something here... and it'll be shipped in spark
   // → care!
 
-  @transient lazy val adam = server.SparkProvider.adamContext
-
-  def index(request:BEACONRequest):BEACONResponse = {
-    new BEACONResponse(true, 1)
+  @transient lazy val sparkContext = server.SparkProvider.sparkContext
+  val source = Sources.`med-at-scale`
+
+  val alleleFreqs = scala.collection.mutable.Map[String, RDD[(Int, Map[Char, Short])]]()
+
+  def index(request: BEACONRequest): BEACONResponse = {
+    val chr = request.getChromosome()
+    val coord = request.getCoordinate()
+    val allele = request.getAllele().charAt(0)
+
+    val rdd = if (alleleFreqs.isDefinedAt(chr)) alleleFreqs(chr)
+              else {
+                val tmprdd: RDD[(Long, Map[String, Long])] = sparkContext.objectFile(source.sourceOfChr(chr))
+                val alls = tmprdd.map{cnt => (cnt._1.toInt, cnt._2.map{all => (all._1.charAt(0) -> all._2.toShort)})}
+                alls.cache()
+                alleleFreqs += (chr -> alls)
+                alls
+              }
+//    val rdd: RDD[(Long, Map[String, Long])] = sparkContext.objectFile(source.sourceOfChr(chr))
+//    val alls = rdd.map{cnt => (cnt._1.toInt, cnt._2.map{all => (all._1.charAt(0) -> all._2.toShort)})}
+//    alls.cache()
+    val variant = rdd.filter(_._1 == coord).take(1)
+    val respOpt = if (variant.size == 1) {
+      val cntOpt: Option[Short] = variant(0)._2.get(allele)
+      cntOpt.map{
+        cnt => 
+          new BEACONResponse(true, (10000.0*cnt/variant(0)._2.values.map(_.toInt).reduce(_+_)).toLong)
```

Comment:
In schema, frequency is a long, I assume it is intended to be a count of seen alleles, not very clear. I compute here a real frequency and multiply by 10,000. Let's make a call on this. (Solved in 0.6.0 as there, frequency is a double as expected).